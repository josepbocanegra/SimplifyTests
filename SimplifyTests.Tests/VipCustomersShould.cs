using System;
using System.Collections.Generic;
using System.Text;
using ApprovalTests;
using ApprovalTests.Reporters;
using ApprovalTests.Xml;
using Newtonsoft.Json;
using Xunit;

namespace SimplifyTests.Tests
{
    [UseReporter(typeof(DiffReporter))]
    public class VipCustomersShould
    {
        [Fact]
        public void ReturnCustomersFullNameInAlphabeticalOrder()
        {
            var vipCustomers = new VipCustomers();
            vipCustomers.Add(new Customer("Mandy", "Harris"));
            vipCustomers.Add(new Customer("John", "Smith"));
            vipCustomers.Add(new Customer("Albert", "Thomson"));

            var orderedVipCustomers = vipCustomers.InAlphabeticalOrder();

            Approvals.VerifyAll(orderedVipCustomers, "");
        }

        [Fact]
        public void ReturnCustomersInAlphabeticalOrderByLastName()
        {
            var vipCustomers = new VipCustomers();
            vipCustomers.Add(new Customer("Albert", "Thomson"));
            vipCustomers.Add(new Customer("Mandy", "Harris"));
            vipCustomers.Add(new Customer("John", "Smith"));
            
            var orderedVipCustomers = vipCustomers.InAlphabeticalOrderByLastName();

            Approvals.VerifyJson(JsonConvert.SerializeObject(orderedVipCustomers));
        }

        [Fact]
        public void GenerateOrderedXmlReport()
        {
            var vipCustomers = new VipCustomers();
            vipCustomers.Add(new Customer("Albert", "Thomson"));
            vipCustomers.Add(new Customer("Mandy", "Harris"));
            vipCustomers.Add(new Customer("John", "Smith"));

            var report = vipCustomers.Report();

            XmlApprovals.VerifyOrderedXml(report);
        }

    }
}
