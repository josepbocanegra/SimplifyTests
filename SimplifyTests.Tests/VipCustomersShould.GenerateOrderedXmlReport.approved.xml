<ArrayOfCustomer xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <Customer>
    <FirstName>Mandy</FirstName>
    <LastName>Harris</LastName>
  </Customer>
  <Customer>
    <FirstName>John</FirstName>
    <LastName>Smith</LastName>
  </Customer>
  <Customer>
    <FirstName>Albert</FirstName>
    <LastName>Thomson</LastName>
  </Customer>
</ArrayOfCustomer>